readthedocs/builds/models.py

```diff
@@ -698,3 +699,114 @@ def run_time(self):
         if self.start_time is not None and self.end_time is not None:
             diff = self.end_time - self.start_time
             return diff.seconds
+
+
+class VersionAutomationRule(PolymorphicModel, TimeStampedModel):
+
+    """Versions automation rules for projects."""
+
+    REGEX_RULE = 'regex-rule'
+    RULE_TYPES = (
+        (REGEX_RULE, _('Regular expression rule')),
+    )
+
+    ACTIVATE_VERSION_ACTION = 'activate-version'
+    ACTIONS = (
+        (ACTIVATE_VERSION_ACTION, _('Activate version on match')),
```

Comment:
Another action could be, `Mark as stable` (in case we support it at #3723) and another `Mark as default version`

Reply:
Mark as default is useful. Mark as stable, we first need to allow marking as stable manually before adding it as an automation rule.

Reply:
Yes. I'd like to have more than 1 action working on the first approach so "multiple actions" are tested from the beginning of the design.